lib/client.js

```diff
@@ -122,9 +143,19 @@ class Client {
 
       log.info('GET: ' + requestOptions.uri)
       request(requestOptions, (error, resp, body) => {
+        // Handle low level, non-http error:
         if (error) reject(error)
-        else if (!body || !body.data) reject('Invalid response: for ' + url, body)
-        else {
+
+        // Handle token expiration:
+        else if (body && body.statusCode === 401) {
```

Comment:
Very good point. I missed that I was checking the wrong object in the response. Just added a commit that checks response.statusCode instead of relying on same in json serialization. Thanks!